plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlabMergeRequest.ts

```diff
@@ -0,0 +1,197 @@
+          },
+          title: {
+            type: 'string',
+            title: 'Merge Request Name',
+            description: 'The name for the merge request',
+          },
+          description: {
+            type: 'string',
+            title: 'Merge Request Description',
+            description: 'The description of the merge request',
+          },
+          destinationBranch: {
+            type: 'string',
+            title: 'Destination Branch name',
+            description: 'The description of the merge request',
+          },
+          targetPath: {
+            type: 'string',
+            title: 'Repository Subdirectory',
+            description: 'Subdirectory of repository to apply changes to',
+          },
+        },
+      },
+      output: {
+        type: 'object',
+        properties: {
+          projectid: {
+            title: 'Gitlab Project id',
+            type: 'string',
+          },
+          mergeRequestURL: {
+            title: 'MergeRequest(MR) URL',
+            type: 'string',
+            description: 'Link to the merge request in GitLab',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const repoUrl = ctx.input.repoUrl;
+      const { host } = parseRepoUrl(repoUrl, integrations);
+      const integrationConfig = integrations.gitlab.byHost(host);
+
+      const actions: CommitAction[] = [];
+      const formatedTimestamp = () => {
+        const d = new Date();
+        const date = d.toISOString().split('T')[0];
+        const time = d.toTimeString().split(' ')[0].replace(/:/g, '_');
+        return `${date}_${time}`;
+      };
+      const destinationBranch = ctx.input.destinationBranch
+        ? ctx.input.destinationBranch + formatedTimestamp()
+        : `backstage_${formatedTimestamp()}`;
+
+      if (!integrationConfig) {
+        throw new InputError(
+          `No matching integration configuration for host ${host}, please check your integrations config`,
+        );
+      }
+
+      if (!integrationConfig.config.token) {
+        throw new InputError(`No token available for host ${host}`);
+      }
+
+      const api = new Gitlab({
+        host: integrationConfig.config.baseUrl,
+        token: integrationConfig.config.token,
+      });
+
+      const fileRoot = ctx.workspacePath;
+      const localFilePaths = await globby([`${ctx.input.targetPath}'/**'`], {
+        cwd: fileRoot,
+        gitignore: true,
+        dot: true,
+      });
+
+      const fileContents = await Promise.all(
+        localFilePaths.map(p => readFile(path.resolve(fileRoot, p))),
+      );
+
+      const repoFilePaths = localFilePaths.map(repoFilePath => {
+        return repoFilePath;
+      });
+
+      for (let i = 0; i < repoFilePaths.length; i++) {
+        actions.push({
+          action: 'create',
+          filePath: repoFilePaths[i],
+          content: fileContents[i].toString(),
+        });
+      }
+
+      const defaultBranch: any = await api.Projects.show(
```

Comment:
Any particular reason to resort to `any` here? are we lacking types?

Reply:
Hi @Rugvip  yes looks like gitbeaker doesnt return the type for the function hence i had to stick to any.

Reply:
Hmm, looks to me like it does? 
![image](https://user-images.githubusercontent.com/4984472/144288144-a5e7d7fd-f4b3-4099-94d8-7050a56bf2bb.png)

Reply:
@Rugvip I have added the changes. but would like your feedback if its what you meant

Reply:
@Rugvip please can you help here when you are available